README.md

```diff
@@ -182,122 +168,80 @@ Your routes file should now look like:
+Update `app/views/restaurants/index.html.erb` so that table column headers are wired to the StimulusReflex action `filter`, passing in the current active filter as an attribute.
 
 ```rb
-module RestaurantsHelper
-  def price_to_dollar_signs(price)
-    "$" * price
-  end
-
-  def stars_to_symbol(stars)
-    "★" * stars
-  end
-end
-```
-
-
-Update `app/views/restaurants/index.html.erb`
-
-```html
-<h1 class="mt-8 text-4xl text-gray-800 font-bold">Restaurants</h1>
-<h2 class="mb-8 text-gray-800">Currently filtering by: <%= session[:filter].capitalize %></h2>
-
-<table class="table-auto bg-white w-full" data-controller="restaurants">
   <thead>
     <tr class="text-left bg-gray-800 text-white">
+      <% FILTERS.each do |filter| %>
       <th class="px-4 py-2">
-        <%= link_to "Name", "#", class: filter_css(:name), data: { reflex: "click->RestaurantsReflex#filter", room: session.id, filter: "name" } %>
-      </th>
-      <th class="px-4 py-2">
-        <%= link_to "Stars", "#", class: filter_css(:stars), data: { reflex: "click->RestaurantsReflex#filter", room: session.id, filter: "stars" } %>
-      </th>
-      <th class="px-4 py-2">
-        <%= link_to "Price", "#", class: filter_css(:price), data: { reflex: "click->RestaurantsReflex#filter", room: session.id, filter: "price" } %>
-      </th>
-      <th class="px-4 py-2">
-        <%= link_to "Category", "#", class: filter_css(:category), data: { reflex: "click->RestaurantsReflex#filter", room: session.id, filter: "category" } %>
+        <%= link_to filter.capitalize, "#", class: filter_css(filter), data: { reflex: "click->RestaurantsReflex#filter", room: session.id, filter: filter } %>
       </th>
+      <% end %>
     </tr>
   </thead>
-
-  <tbody class="text-gray-900">
-    <% @filtered_restaurants.each do |restaurant| %>
-      <tr>
-        <td class="border px-4 py-2"><%= restaurant.name %></td>
-        <td class="border px-4 py-2"><%= stars_to_symbol(restaurant.stars) %></td>
-        <td class="border px-4 py-2"><%= price_to_dollar_signs(restaurant.price) %></td>
-        <td class="border px-4 py-2"><%= restaurant.category %></td>
-      </tr>
-    <% end %>
-  </tbody>
-</table>
 ```
 
 ### 10
 
-Create `app/reflexes/restaurants_reflex.rb`
+Create `filter` Reflex method.. The sorting order is `:normal` unless the user clicks again on the same column heading multiple times.
 
 ```rb
+# app/reflexes/restaurants_reflex.rb
 class RestaurantsReflex < StimulusReflex::Reflex
   def filter
-    session[:filter] = element.dataset[:filter]
     session[:filter_order] = filter_order
+    session[:filter] = element.dataset[:filter]
   end
 
   private
 
   def filter_order
-    return :reverse if session[:filter] == element.dataset[:filter] && session[:filter_order] != :reverse
-
-    :normal
+    return :normal unless session[:filter] == element.dataset[:filter]
+    if session[:filter_order] != :reverse
+      :reverse
+    else
+      :normal
+    end
```

Comment:
🎉

Reply:
So, yeah: I kind of like the way that you did it better.

I was trying to be "more clear" but yours is both way shorter and I'm not convinced it's *unclear*.